src/App.tsx

```diff
@@ -1,12 +1,95 @@
 import { ThemeProvider, ensure, themes } from "@storybook/theming";
-import React from "react";
+import { type API } from "@storybook/manager-api";
 
 const theme = ensure(themes.light);
 
-export function App() {
+import React, { useCallback, useEffect, useState } from "react";
+import { STORY_CHANGED, CURRENT_STORY_WAS_SET } from "@storybook/core-events";
+import { GuidedTour } from "./features/GuidedTour/GuidedTour";
+import { WelcomeModal } from "./features/WelcomeModal/WelcomeModal";
+import { WriteStoriesModal } from "./features/WriteStoriesModal/WriteStoriesModal";
+
+export const App = ({ api }: { api: API }) => {
+  const [enabled, setEnabled] = useState(true);
+  const [showGuidedTour, setShowGuidedTour] = useState(false);
+  const [showWriteStoriesModal, setShowWriteStoriesModal] = useState(false);
+  const [showWelcomeModal, setShowWelcomeModal] = useState(true);
+  const [isFinalStep, setIsFinalStep] = useState(false);
```

Comment:
```suggestion
  type Step = "1:Welcome" | "2:..." | ...
  const [step, setStep] = useState<Step>("1:Welcome");
```

Why do we need so may states? They partially depend on each other, so keeping them in sync is error-prone. Let's use one single state instead. :)

Reply:
I love it!